### PR TITLE
Fixes #33831 - Cancel recurring config job

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: webpack/components/AnsibleHostDetail/components/JobsTab/JobsTabHelper.js [hunk 1]
```diff
@@ -24,3 +27,40 @@ export const fetchPreviousFn = fetchJobsFn(previousJobsSearch);
...
+    __('There was a following error when deleting Ansible config job: %s'),
+    error
+  );
+
+const onError = error => {
+  showToast({ type: 'danger', message: formatError(error) });
+};
+
+const onCompleted = data => {
+  const { errors } = data.cancelRecurringLogic;
```
Comment (amirfefer) [inline webpack/components/AnsibleHostDetail/components/JobsTab/JobsTabHelper.js:44]: can`cancelRecurringLogic` be undefeind?

File: webpack/components/AnsibleHostDetail/components/JobsTab/RecurringJobsTable.js [hunk 2]
```diff
@@ -28,11 +44,12 @@ const RecurringJobsTable = props => {
...
               <Th key={col}>{col}</Th>
             ))}
+            <Th />
           </Tr>
         </Thead>
         <Tbody>
-          {props.jobs.map(job => (
-            <Tr key={job.id}>
+          {jobs.map((job, idx) => (
+            <Tr key={idx}>
```
Comment (amirfefer) [inline webpack/components/AnsibleHostDetail/components/JobsTab/RecurringJobsTable.js:52]: isn't the job's id good enough here?

**Review by amirfefer — COMMENTED** [review]
overall code looks 👍 
can't test it due to my env though

File: webpack/components/AnsibleHostDetail/components/JobsTab/JobsTabHelper.js (hunk 1 above)
Comment (xprazak2) [reply webpack/components/AnsibleHostDetail/components/JobsTab/JobsTabHelper.js:44]: I do not think so, that is what is always sent when mutation completes - unless there is network error which is handled by `onError`

File: webpack/components/AnsibleHostDetail/components/JobsTab/RecurringJobsTable.js (hunk 2 above)
Comment (xprazak2) [reply webpack/components/AnsibleHostDetail/components/JobsTab/RecurringJobsTable.js]: Seems like rebase leftovers, updated.

**Comment by ezr-ondrej** [comment]
I'm unable to schedule the recurring job now :( (I'm not sure if it's related, but I can't test cancel as I have in-memory dynflow and it will fail to cancel jobs created in another run)

**Comment by xprazak2** [comment]
Ok, there should no longer be errors when you try scheduling a new job.

**Review by ezr-ondrej — APPROVED** [review]
Works as a charm, thanks @xprazak2 !

**Pull request merged**